executor/executor.go

```diff
@@ -533,3 +500,77 @@ func (e *Executor) throwError(taskID mesos.TaskID, err error) error {
 
 	return err
 }
+
+// tearDownTask kills the given task, running hooks and stopping associated container before
+// updating the task status
+func (e *Executor) tearDownTask(taskID mesos.TaskID, containerTaskInfo *types.ContainerTaskInfo) error {
+	// Quit and remove health checker (if existing)
+	e.HealthCheckersMutex.Lock()
+	if hc, ok := e.HealthCheckers[taskID]; ok {
+		hc.Quit <- struct{}{}
+	}
+
+	delete(e.HealthCheckers, taskID)
+	e.HealthCheckersMutex.Unlock()
+
+	// Run pre-stop hooks
+	err := e.HookManager.RunPreStopHooks(e.Containerizer, containerTaskInfo)
+	if err != nil {
+		return err
+	}
+
+	// Stop container
+	err = e.Containerizer.ContainerStop(containerTaskInfo.ContainerID)
```

Comment:
we may run `tearDownTask` multiple time concurrently. We might pill up calls to container STOP here.

I do not know the behaviour of https://godoc.org/github.com/fsouza/go-dockerclient#Client.StopContainer if you try to stop a container already stopped.

Reply:
Docker library StopContainer function (called by containerizer interface) is idempotent on container stop and will not throw an error if the container is already stopped.